src/main/java/com/example/bookstoreapp/service/impl/BookServiceImpl.java

```diff
@@ -48,17 +48,19 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto getBookById(Long id) {
-        Book book = bookRepository.findById(id).orElseThrow(
+        return bookRepository.findById(id).map(bookMapper::toDto).orElseThrow(
                 () -> new EntityNotFoundException(CANT_FIND_BOOK_MSG + id));
-        return bookMapper.toDto(book);
     }
 
     @Override
     public List<BookDto> searchBooks(BookSearchParametersDto bookSearchParametersDto) {
         Specification<Book> bookSpecification = specificationBuilder
                 .build(bookSearchParametersDto);
         List<Book> bookList = bookRepository.findAll(bookSpecification);
-        return bookMapper.toDtos(bookList);
+        if (!bookList.isEmpty()) {
+            return bookMapper.toDtos(bookList);
+        }
+        throw new EntityNotFoundException("Books not found ");
```

Comment:
```suggestion
          return bookMapper.toDtos(bookList);
```
return empty collection with 200 response status